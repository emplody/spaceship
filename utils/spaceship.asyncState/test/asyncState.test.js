import asyncState, { PENDING, SUCCESS, FAILURE } from '../index';

test('asyncState return pending true', () => {
  const { pending, success, failure } = asyncState(PENDING);
  expect(pending).toBeTruthy();
  expect(success).toBeFalsy();
  expect(failure).toBeFalsy();
});

test('asyncState return success true', () => {
  const { pending, success, failure } = asyncState(SUCCESS);
  expect(pending).toBeFalsy();
  expect(success).toBeTruthy();
  expect(failure).toBeFalsy();
});

test('asyncState return pending true', () => {
  const { pending, success, failure } = asyncState(FAILURE);
  expect(pending).toBeFalsy();
  expect(success).toBeFalsy();
  expect(failure).toBeTruthy();
});

test('if accept unexpected arguments, will return all state false.', () => {
  expect(asyncState('some parameter')).toMatchSnapshot();
});
