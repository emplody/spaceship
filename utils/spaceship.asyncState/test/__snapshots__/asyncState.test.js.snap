// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if accept unexpected arguments, will return all state false. 1`] = `
Object {
  "failure": false,
  "pending": false,
  "success": false,
}
`;
